packetbeat/beater/processor.go

```diff
@@ -0,0 +1,142 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package beater
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/cfgfile"
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/logp"
+	"github.com/elastic/beats/v7/libbeat/publisher/pipeline"
+
+	"github.com/elastic/beats/v7/packetbeat/flows"
+	"github.com/elastic/beats/v7/packetbeat/procs"
+	"github.com/elastic/beats/v7/packetbeat/protos"
+	"github.com/elastic/beats/v7/packetbeat/publish"
+	"github.com/elastic/beats/v7/packetbeat/sniffer"
+)
+
+type processor struct {
+	wg      sync.WaitGroup
+	flows   *flows.Flows
+	sniffer *sniffer.Sniffer
+	err     chan error
+}
+
+func newProcessor(flows *flows.Flows, sniffer *sniffer.Sniffer, err chan error) *processor {
```

Comment:
Who's responsible for closing the err channel? Normally the writer closes it when it's done.

Reply:
Looks like it doesn't matter whether it gets closed in this case as the reader doesn't need it to be closed to stop reading. So nothing to worry about here.

Reply:
The intention was actually to use this channel as a long-lived channel in the goroutine that initializes the factory (which is also the reader goroutine) and use it to propagate errors from whatever new `processor`s might get initialized due to configuration changes.

It looks like I actually forgot to close it 😅 , but so long as no more processors can get created after the channel is closed, which there shouldn't be, then we won't have issues with writing to a closed channel.